### PR TITLE
JP-3081 NIRSpec Imprints improvement - constrain by observation number

### PR DESCRIPTION
**Pull request by jemorrison**
Resolves [JP-3081](https://jira.stsci.edu/browse/JP-3081)

This PR ensures that for an imprint to be subtracted from either a background or science image the observation number and dither position number must match.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by jemorrison** [comment]
Going to open a JP ticket when --imprint_subtract.save_results=True . Currently imprint subtracted background ground images are not written out.

File: jwst/imprint/imprint_step.py [hunk 1]
```diff
@@ -53,8 +55,10 @@ def process(self, input, imprint):
 
             # Update the step status and close the imprint model
             result.meta.cal_step.imprint = 'COMPLETE'
-            input_model.close()
             imprint_model.close()
         else:
             self.log.info(f'No imprint image was found for {input}')
+            result.meta.cal_step.imprint = 'SKIP'
```
Comment (hbushouse) [inline jwst/imprint/imprint_step.py:61]: This should be set to "SKIPPED".

**Review by hbushouse — COMMENTED** [review]
Looks OK to me, other than the one minor update.

**Comment by hbushouse** [comment]
> Going to open a JP ticket when --imprint_subtract.save_results=True . Currently imprint subtracted background ground images are not written out.

Agree that that issue can be fixed separately and worked later (not as important as getting the actual subtraction worked out correctly).

**Review by hbushouse — APPROVED** [review]
LGTM

**Pull request merged**